File: android/src/main/kotlin/com/syncrhrohealth/bandpass_ble_connection/plugin/devicecore/enum/DeviceCoreMethod.kt
package com.syncrhrohealth.bandpass_ble_connection.plugin.devicecore.enum

/** Make sure the method value matches the DeviceCoreMethod on Flutter side */
enum class DeviceCoreMethod(val value: String) {
    // Permissions
    IS_BLE_ENABLED("isBleEnabled"),

    // Scanning
    START_SCAN("startScan"),
    STOP_SCAN("stopScan"),

    // Connection
    CONNECT("connect"),
    DISCONNECT("disconnect"),
    RECONNECT("reconnect"),
    RECONNECT_DEVICES("reconnectDevices"),

    // Device commands
    GET_DEVICE_INFO("getDeviceInfo"),
    SET_TIME_STAMP("setTimeStamp"),
    GET_TIME_STAMP("getTimeStamp"),
}

File: ios/Classes/Plugin/DeviceCore/Enum/DeviceCoreMethod.swift
//
//  Untitled.swift
//  Runner
//
//  Created by MAC on 12/3/25.
//

import Foundation

/// Make sure the method value matches the `DeviceCoreMethod` on the Flutter side
enum DeviceCoreMethod: String {
    // Permissions
    case isBleEnabled = "isBleEnabled"
    case hasBlePermission = "hasBlePermission"

    // Scanning
    case startScan = "startScan"
    case stopScan = "stopScan"

    // Connection
    case connect = "connect"
    case disconnect = "disconnect"
    case reconnect = "reconnect"
    case reconnectDevices = "reconnectDevices"

    // Device commands
    case getDeviceInfo = "getDeviceInfo"
}
